main.cc

```diff
@@ -357,12 +357,39 @@ int main(int ac, char** av) {
     // If --version is requested, print it out and exit immediately to avoid
     // Seastar-specific warnings that may occur when running the app
     bpo::variables_map vm;
-    bpo::store(bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run(), vm);
+    auto parsed_opts = bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run();
+    bpo::store(parsed_opts, vm);
     if (vm["version"].as<bool>()) {
         fmt::print("{}\n", scylla_version());
         return 0;
     }
 
+    if (ac) {
```

Comment:
How can argc possibly be zero? It's always >= 1, and argv[0] is the program's name.

Reply:
It could be zero theoretically, in which case argv[0] is the NULL pointer. The standard only says that if argc > 0, then argv[0] is some string representing the program.
https://en.cppreference.com/w/cpp/language/main_function
https://stackoverflow.com/a/49817618